polkadot/node/network/gossip-support/src/lib.rs

```diff
@@ -259,7 +281,12 @@ where
 							// to clean up all connections.
 							Vec::new()
 						};
-					self.issue_connection_request(sender, connections).await;
+
+					if force_request || is_new_session {
+						self.issue_connection_request(sender, connections).await;
+					} else if re_resolve_authorities {
+						self.issue_connection_request_to_changed(sender, connections).await;
+					}
```

Comment:
Yeah, I'm more concerned that the other way around we may end-up calling re_resolve path instead of issue_connection_request at the first block in the session. But, I agree this is not obvious we don't affect the previous path I'll make it cleaner.